Failing test: test_round_robin_web_test_helpers

[This test](https://github.com/ooni/backend/blob/master/api/tests/unit/test_probe_services.py) is failing with the following error: 
```
AssertionError: assert [{'address': ...pe': 'https'}] == [{'address': ...pe': 'https'}]
E         At index 3 diff: {'address': 'https://4.th.ooni.org', 'type': 'https'} != {'address': 'https://0.th.ooni.org', 'type': 'https'}
E         Left contains one more item: {'address': 'https://0.th.ooni.org', 'type': 'https'}
```

When we look at the section failing: 
```
@patch("ooniapi.probe_services.extract_probe_ipaddr")
def test_round_robin_web_test_helpers(mock):
    # test fallback without mocked ipaddr
    li = ooniapi.probe_services.round_robin_web_test_helpers()
    assert li == [
        {"address": "https://1.th.ooni.org", "type": "https"},
        {"address": "https://2.th.ooni.org", "type": "https"},
        {"address": "https://3.th.ooni.org", "type": "https"},
        {"address": "https://0.th.ooni.org", "type": "https"},
    ]
...
```

And check the [code tested](https://github.com/ooni/backend/blob/46d5d7297f2dec989a844ea5d1247ec8fc72779c/api/ooniapi/probe_services.py#L515): 
```
def round_robin_web_test_helpers() -> List[Dict]:
    """Round robin test helpers based on the probe ipaddr.
    0.th is special and gets only 10% of the traffic.
    """
    try:
        ipa = extract_probe_ipaddr()
        # ipaddr as (large) integer representation (v4 or v6)
        q = int(ipaddress.ip_address(ipa))
        q = q % 100
    except Exception:
        q = 12  # pick 1.th

    if q < 10:
        shift = 0
    else:
        shift = q % 4 + 1

    out = []
    # V Here is the problem V
    for n in range(5):
        n = (n + shift) % 5
        out.append({"address": f"https://{n}.th.ooni.org", "type": "https"})

    return out
```

We note that the output list will always contain at the least one more element that the expected list in the test 

We should either update the test or check if the output of the function is expected

--- a/api/ooniapi/probe_services.py
+++ b/api/ooniapi/probe_services.py
@@ -527,11 +527,11 @@ def round_robin_web_test_helpers() -> List[Dict]:
     if q < 10:
         shift = 0
     else:
-        shift = q % 4 + 1
+        shift = q % 3 + 1
 
     out = []
-    for n in range(5):
-        n = (n + shift) % 5
+    for n in range(4):
+        n = (n + shift) % 4
         out.append({"address": f"https://{n}.th.ooni.org", "type": "https"})
 
     return out

